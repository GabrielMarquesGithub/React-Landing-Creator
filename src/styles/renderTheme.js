import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (element) => {
  return render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
};
