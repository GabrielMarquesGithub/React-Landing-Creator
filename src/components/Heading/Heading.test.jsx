import { renderTheme } from '../../styles/renderTheme';
import { Heading } from '.';
import { screen } from '@testing-library/react';
import { theme } from '../../styles/theme';
import 'jest-styled-components';

describe('<Heading/>', () => {
  it('Should render Heading component with default values', () => {
    renderTheme(<Heading>text</Heading>);
    const heading = screen.getByRole('heading', { name: 'text' });

    //teste valores padrÃ£o
    expect(heading).toHaveStyle({
      color: theme.colors.primary,
      'font-size': '6.4rem',
      'text-transform': 'none',
    });
  });

  it('Should render with white color', () => {
    renderTheme(<Heading colorDark={false}>text</Heading>);
    //selecionando o elemento heading
    const heading = screen.getByRole('heading', { name: 'text' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('Should render small Heading', () => {
    renderTheme(<Heading size="small">text</Heading>);
    const heading = screen.getByRole('heading', { name: 'text' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.medium,
    });
  });
  it('Should render medium Heading', () => {
    renderTheme(<Heading size="medium">text</Heading>);
    const heading = screen.getByRole('heading', { name: 'text' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.large,
    });
  });
  it('Should render big Heading', () => {
    renderTheme(<Heading size="big">text</Heading>);
    const heading = screen.getByRole('heading', { name: 'text' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.xlarge,
    });
  });
  it('Should render huge Heading', () => {
    renderTheme(<Heading size="huge">text</Heading>);
    const heading = screen.getByRole('heading', { name: 'text' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.xhuge,
    });
  });

  it('Should render correct font-size when using mobile', () => {
    renderTheme(<Heading size="huge">text</Heading>);
    const heading = screen.getByRole('heading', { name: 'text' });

    expect(heading).toHaveStyleRule('font-size', theme.font.size.xlarge, {
      media: `(${theme.media.lessMedium})`,
    });
  });

  it('Should render with uppercase letters', () => {
    renderTheme(<Heading uppercase={true}>text</Heading>);
    const heading = screen.getByRole('heading', { name: 'text' });

    expect(heading).toHaveStyle({ 'text-transform': 'uppercase' });
  });

  it('Should render correct heading', () => {
    const { container } = renderTheme(<Heading as="h6">text</Heading>);
    const h6 = container.querySelector('h6');

    expect(h6.tagName.toLowerCase()).toBe('h6');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading>text</Heading>);
    expect(container).toMatchSnapshot();
  });
});
