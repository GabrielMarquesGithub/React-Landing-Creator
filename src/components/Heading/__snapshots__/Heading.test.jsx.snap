// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> should match snapshot 1`] = `
.c0 {
  text-transform: none;
  color: #0A1128;
  font-size: 6.4rem;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<div>
  <h1
    class="c0"
  >
    text
  </h1>
</div>
`;
